datashuttle/command_line_interface.py

```diff
@@ -0,0 +1,662 @@
+make_config_file_parser.add_argument(
+    "--remote_host_username", required=False, help="(str)", metavar=""
+)
+make_config_file_parser.add_argument(
+    "--sub_prefix",
+    required=False,
+    type=str,
+    metavar="",
+    help="(str) default: sub-",
+)
+make_config_file_parser.add_argument(
+    "--ses_prefix",
+    required=False,
+    type=str,
+    metavar="",
+    help="(str) default: ses-",
+)
+make_config_file_parser.add_argument(
+    "--use_ephys",
+    required=False,
+    default=None,
+    action="store",
+    metavar="",
+    help=help("flag_default_false"),
+)
+make_config_file_parser.add_argument(
+    "--use_ephys_behav",
+    required=False,
+    default=None,
+    action="store",
+    metavar="",
+    help=help("flag_default_false"),
+)
+make_config_file_parser.add_argument(
+    "--use_ephys_behav_camera",
+    required=False,
+    default=None,
+    action="store",
+    metavar="",
+    help=help("flag_default_false"),
+)
+make_config_file_parser.add_argument(
+    "--use_behav",
+    required=False,
+    default=None,
+    action="store",
+    metavar="",
+    help=help("flag_default_false"),
+)
+make_config_file_parser.add_argument(
+    "--use_behav_camera",
+    required=False,
+    default=None,
+    action="store",
+    metavar="",
+    help=help("flag_default_false"),
+)
+make_config_file_parser.add_argument(
+    "--use_imaging",
+    required=False,
+    default=None,
+    action="store",
+    metavar="",
+    help=help("flag_default_false"),
+)
+make_config_file_parser.add_argument(
+    "--use_histology",
+    required=False,
+    default=None,
+    action="store",
+    metavar="",
+    help=help("flag_default_false"),
+)
+
+
+# ------------------------------------------------------------------------------------------
+# Update Config
+# ------------------------------------------------------------------------------------------
+
+
+def update_config(project, args):
+    """"""
+    kwargs = make_kwargs(args)
+    option_key = kwargs["option_key"]
+    new_info = kwargs["new_info"]
+
+    new_info = handle_bool(option_key, new_info)
+
+    run_command(
+        project,
+        project.update_config,
+        option_key,
+        new_info,
+    )
+
+
+make_config_file_parser = subparsers.add_parser(
+    "update_config",
+    description=DataShuttle.update_config.__doc__,
+    formatter_class=argparse.RawTextHelpFormatter,
+    help="",
+)
+make_config_file_parser.set_defaults(func=update_config)
+
+make_config_file_parser.add_argument(
+    "option_key",
+    action="store",
+    help="(str) (see make_config_file --help)",
+)
+make_config_file_parser.add_argument(
+    "new_info", action="store", help="(str or bool) depending on option key"
+)
+
+
+# ------------------------------------------------------------------------------------------
+# Setup SSH
+# ------------------------------------------------------------------------------------------
+
+
+def setup_ssh_connection_to_remote_server(project, args):
```

Comment:
why is args here if not accessed

Reply:
Yes I wasn't sure about this, argparse passes the args with the linked .func attached, which could be any of the run functions involved in running CLI commands (e.g. check_name_processing). Couldnt figure out how to handle this nicely, but now used *args in these cases